### ISSUE TITLE
Map with constructor

### ISSUE THREAD

**Issue by wehrstedt** [amilajack/eslint-plugin-compat#248]
I'm using `Map` in my client code. I passed an array to the constructor as it is supported in several runtimes, but not in IE.

This does not work, the map will be empty:
```const map = new Map(["fuu", "bar"]);```

There is no linting error of this plugin. This is my configuration:
```
{
	"extends": [
		"plugin:compat/recommended",
		"../../../.eslintrc.json"
	],
	"plugins": [
		"es"
	],
	"env": {
		"browser": true
	},
	"overrides": [
		{
			"files": [
				"**/clientFunctions/**/*.js"
			],
			"rules": {
				"es/no-arrow-functions": ["error"],
				"es/no-for-of-loops": ["error"],
				"es/no-rest-spread-properties": ["error"],
				"es/no-spread-elements": ["error"]
			}
		}
	]
}
```

And the .browserlistrc
```json
ie 11
```

Expected to get a linting error as Map with constructor parameter is not supported in IE11

**Comment by naruaway** [comment]
I think it's the same situation for `new Set(iterable)`(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set)

**Comment by brettz9** [comment]
I think this should be closed in favor of the more generic #206 (letting that apply to all ES features).
